haystack/components/preprocessors/document_preprocessor.py

```diff
@@ -0,0 +1,181 @@
+            `respect_sentence_boundary=True`.
+        :param use_split_rules: Whether to apply additional splitting heuristics for the sentence splitter.
+        :param extend_abbreviations: Whether to extend the sentence splitter with curated abbreviations for certain
+            languages.
+        """
+        # Store arguments for serialization
+        self.remove_empty_lines = remove_empty_lines
+        self.remove_extra_whitespaces = remove_extra_whitespaces
+        self.remove_repeated_substrings = remove_repeated_substrings
+        self.keep_id = keep_id
+        self.remove_substrings = remove_substrings
+        self.remove_regex = remove_regex
+        self.unicode_normalization = unicode_normalization
+        self.ascii_only = ascii_only
+
+        self.split_by = split_by
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_threshold = split_threshold
+        self.splitting_function = splitting_function
+        self.respect_sentence_boundary = respect_sentence_boundary
+        self.language = language
+        self.use_split_rules = use_split_rules
+        self.extend_abbreviations = extend_abbreviations
+
+        # Instantiate sub-components
+        cleaner = DocumentCleaner(
+            remove_empty_lines=self.remove_empty_lines,
+            remove_extra_whitespaces=self.remove_extra_whitespaces,
+            remove_repeated_substrings=self.remove_repeated_substrings,
+            keep_id=self.keep_id,
+            remove_substrings=self.remove_substrings,
+            remove_regex=self.remove_regex,
+            unicode_normalization=self.unicode_normalization,
+            ascii_only=self.ascii_only,
+        )
+
+        splitter = DocumentSplitter(
+            split_by=self.split_by,
+            split_length=self.split_length,
+            split_overlap=self.split_overlap,
+            split_threshold=self.split_threshold,
+            splitting_function=self.splitting_function,
+            respect_sentence_boundary=self.respect_sentence_boundary,
+            language=self.language,
+            use_split_rules=self.use_split_rules,
+            extend_abbreviations=self.extend_abbreviations,
+        )
+
+        # Build the Pipeline
+        pp = Pipeline()
+        pp.add_component("cleaner", cleaner)
+        pp.add_component("splitter", splitter)
+
+        # Connect the splitter output to cleaner
+        pp.connect("splitter.documents", "cleaner.documents")
+        self.pipeline = pp
+
+        # Define how pipeline inputs/outputs map to sub-component inputs/outputs
+        self.input_mapping = {
+            # The pipeline input "documents" feeds into "splitter.documents"
+            "documents": ["splitter.documents"]
+        }
+        # The pipeline output "documents" comes from "cleaner.documents"
+        self.output_mapping = {"cleaner.documents": "documents"}
+
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
(I did not take a look at https://github.com/deepset-ai/haystack/pull/9233)

Could you please explain why this `SuperComponent` needs `to_dict`/`from_dict` methods while `MultiFileConverter` does not?

Reply:
This one needs it since it has a non-standard serialization. By that I mean it needs to serialize the splitting_function which isn't supported by `default_to_dict` nor `component_to_dict`. 

The other one has no special needs, so it is able to use the default `component_to_dict` like our other components.